lib/periodic.ex

```diff
@@ -127,6 +146,7 @@ defmodule Periodic do
   @impl GenServer
   def init(opts) do
     state = defaults() |> Map.merge(opts) |> Map.put(:timer, nil)
+    enqueue_initial(state)
```

Comment:
Since you invoke `enqueue_initial` and `enqueue_next`, we'll end up with two interleaved executions. I propose a slightly different approach:

```elixir
{initial_delay, opts} = Keyword.pop(opts, :initial_delay, Keyword.fetch!(opts, :every))
state = defaults() |> Map.merge(opts) |> Map.put(:timer, nil)
enqueue_initial(initial_delay)
{:ok, state}
```